clang/test/CIR/Lowering/loops-with-break.cir

```diff
@@ -0,0 +1,447 @@
+// RUN: cir-opt %s -cir-to-llvm -o %t.mlir
+// RUN: FileCheck --input-file=%t.mlir %s -check-prefix=MLIR
+
+!s32i = !cir.int<s, 32>
+module {
+  cir.func @testFor() {
+    cir.scope {
+      %0 = cir.alloca !s32i, cir.ptr <!s32i>, ["i", init] {alignment = 4 : i64}
+      %1 = cir.const(#cir.int<1> : !s32i) : !s32i
+      cir.store %1, %0 : !s32i, cir.ptr <!s32i>
+      cir.loop for(cond : {
+        %2 = cir.load %0 : cir.ptr <!s32i>, !s32i
+        %3 = cir.const(#cir.int<10> : !s32i) : !s32i
+        %4 = cir.cmp(lt, %2, %3) : !s32i, !s32i
+        %5 = cir.cast(int_to_bool, %4 : !s32i), !cir.bool
+        cir.brcond %5 ^bb1, ^bb2
+      ^bb1:  // pred: ^bb0
+        cir.yield continue
+      ^bb2:  // pred: ^bb0
+        cir.yield
+      }, step : {
+        %2 = cir.load %0 : cir.ptr <!s32i>, !s32i
+        %3 = cir.unary(inc, %2) : !s32i, !s32i
+        cir.store %3, %0 : !s32i, cir.ptr <!s32i>
+        cir.yield
+      }) {
+        cir.scope {
+          cir.scope {
+            %2 = cir.load %0 : cir.ptr <!s32i>, !s32i
+            %3 = cir.const(#cir.int<5> : !s32i) : !s32i
+            %4 = cir.cmp(eq, %2, %3) : !s32i, !s32i
+            %5 = cir.cast(int_to_bool, %4 : !s32i), !cir.bool
+            cir.if %5 {
+              cir.yield break
+            }
+          }
+        }
+        cir.yield
+      }
+    }
+    cir.return
+  }
+
+// MLIR:          llvm.func @testFor()
```

Comment:
@gitoleg would you mind simplifying the tests to check for _only_ the branching between blocks?

It helps keep the tests more focused and change-proof. We often have tests breaking due to unrelated changes when `// CHECK:` directives are applied to the entirety of the output code.

Here's an [example](https://github.com/llvm/clangir/pull/178/commits/c48defed0fee112b06702aebe786eae74f5dad92#diff-01bd60b04e765266c60d0bdef5730b0525043fd1854b7cc3ddd854d16e212678) of a "branch-only" validation.

Reply:
@sitio-couto  take a look please